### ISSUE TITLE
Test failing with Java exception linked to SSL issue?

### ISSUE THREAD

**Issue by mackrauss** [iNPUTmice/ComplianceTester#43]
Not sure what is happening here, but when I try to test my Prosody server I get the following error (see file):

[java-error.txt](https://github.com/iNPUTmice/ComplianceTester/files/573359/java-error.txt)

What can I do to still be able to test my prosody server? Would love to start using Conversations app with all features, but don't know where I currently stand.

Thanks for any help and pointers.

**Comment by iNPUTmice** [comment]
> Would love to start using Conversations app with all features, but don't know where I currently stand.

In the menu of the account details in Conversations you'll find 'Show server info'. Use this to quickly determine if your server supports the important features.

The compliance tester can't handle self signed certificates and/or your java installation needs access to the certificate store on your OS. This won't change any time soon.

**Comment by andreygursky** [comment]
@mackrauss, for a Java program this can be done as follows:
* for your local prosody: as root ```cp -ai /etc/prosody/certs/your-domain.crt /path/to/your/working/dir/domain.crt; chown YOUR-USERNAME /path/to/your/working/dir/domain.crt```
* (or for a server with a bad certificate (like jabber.ccc.de at the moment) ```echo | openssl s_client -connect jabber.ccc.de:443 2>/dev/null | openssl x509 > /path/to/your/working/dir/domain.crt```)
* ```keytool -import -alias prosody -file /path/to/your/working/dir/domain.crt```
   (enter some password)
* check the alias has been added: ```keytool -list```
* run java program with additional arguments: ```-Djavax.net.ssl.trustStore=/home/YOUR-USERNAME/.keystore -Djavax.net.ssl.keyStorePassword=(here the previously entered password)```

**Comment by pintman** [comment]
Is it possible to accept self signed certificates for only one invocation of the tool?

**Comment by andreygursky** [comment]
Yes. Proceed as describe above. Once you don't need the self signed certificate, don't pass arguments during invocation. If you don't need the self signed certificate anymore, you could delete the keystore file (safe, if there was no such file before you've added the first certificate).

**Comment by owhen** [comment]
I have the same issue, but I'm using a trusted Lets Encrypt certificate. The error occurs if I run this jar file for a second virtual vhost (prosody).

**Comment by vv01f** [comment]
@andreygursky 
Soweit funktioniert das; aber gerade für den erwähnten jabber.ccc.de kommt dann:
```
Use compliance suite 'Conversations Compliance Suite' to test jabber.ccc.de

The ComplianceTester can not handle DH key sizes above 2048 bit. Modify your Prosody to use the default TLS configuration.
```
Der Handlungsanweisung kann ein Nutzer leider nicht folgen; gibt es einen Weg darum herum?
